x/escrow/keeper/keeper.go

```diff
@@ -550,31 +583,32 @@ func accountSettleFullblocks(
 
 func accountSettleDistributeWeighted(
 	account types.Account,
-	payments []types.Payment,
-	blockRate sdk.Coin,
-	amountRemaining sdk.Coin,
+	payments []types.FractionalPayment,
+	blockRate sdk.DecCoin,
+	amountRemaining sdk.DecCoin,
 ) (
 	types.Account,
-	[]types.Payment,
-	sdk.Coin,
+	[]types.FractionalPayment,
+	sdk.DecCoin,
 ) {
 
-	actualTransferred := sdk.ZeroInt()
+	actualTransferred := sdk.ZeroDec()
 
 	// distribute remaining balance weighted by rate
+
 	for idx := range payments {
 		payment := payments[idx]
 		// amount := (rate / blockrate) * remaining
 		amount := amountRemaining.Amount.
 			Mul(payment.Rate.Amount).
```

Comment:
uh... does this mutate `amountRemaining.Amount`?

Reply:
I don't actually know, I'll have to look at the implementation.